test/cpp-raw/gcc/os.cpp

```diff
@@ -0,0 +1,35 @@
+
+#include <unordered_map>
+#include <mutex>
+
+#include "../os.h"
+#include "test_lib.h"
+#include "../../../intercom-cpp/src/cominterop.h"
+#include "../../../intercom-cpp/src/activator.h"
+
+using intercom::cpp::Activator;
+
+// Storage for libraries.
+namespace
+{
+	static std::mutex m_libraryLock;
+	static std::unordered_map<std::string, dlopen_wrapper > g_libraries;
+}
```

Comment:
I started considering it but then re-focused on the interface side. Something like this will be  included in the code generated from the Rust library and this can be removed then.